Refactor, improve the WebAPIClient tutorial page to use `GetFromJsonAsync` instead of `DeserializeAsync`

BartoszKlonowski opened the pull request:
This pull request closes #47386.

It uses the `System.Net.Http.Json.HttpClientJsonExtensions.GetFromJsonAsync` extension method to simplify the fetching and deserialization process in the tutorial to teach good practices from the beginning and use the latest available features.

To achieve the full compatibility with what was there in the tutorial and the new method usage, this PR:
* Refactors the code of `ProcessRepositoriesAsync` function to make use of the `GetFromJsonAsync` method,
* Refactors the `Repository` class so that it doesn't have to use the complicated `JsonPropertyName` of `System.Text.Json.Serialization`, but puts all the class fields that we want to deserialize right away into the class fields,
* Changes the part of "Deserialize the JSON Result" so that the new method is described,
* Removes completely the part deserialization configuration - it was just not needed anymore considering the simplified approach,
* Updates snippets so that they match what is implemented step-by-step in the article,
* Changes `Write` to `WriteLine`, so that at the beginning, when displaying only the repo name in the console we display it line by line, instead of all together (to slightly improve the readability).
* Modifies the numeration of the points in the tutorials.
I know this displays correctly already in the final page in docs, but it just confused me at first, so wanted to align that with what is displayed in page.

**Note**
This PR should also have the PR to samples repo merged: https://github.com/dotnet/samples/pull/7067, to make sure that the full, final, project matches the tutorial implementation explained.

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/tutorials/console-webapiclient.md](https://github.com/dotnet/docs/blob/9dbc288cc178ab2c3062054362153a89c1f4dfa6/docs/csharp/tutorials/console-webapiclient.md) | [Tutorial: Make HTTP requests in a .NET console app using C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/tutorials/console-webapiclient?branch=pr-en-us-48202) |


<!-- PREVIEW-TABLE-END -->

BartoszKlonowski committed:
Use Net.Http.Json extension method GetFromJsonAsync instead of JsonSerializer

BartoszKlonowski committed:
Remove configuring the deserialization

BartoszKlonowski committed:
Capitalize the Name property

BartoszKlonowski committed:
Update Repository class

BartoszKlonowski committed:
Remove unused Text.Json.Serialization

BartoszKlonowski committed:
Correct the orders of points

BartoszKlonowski committed:
WriteLine instead of Write in early phase to improve output readability

BillWagner commented on docs/csharp/tutorials/console-webapiclient.md:
```diff
@@ -121,37 +121,33 @@ Use the <xref:System.Net.Http.HttpClient> class to make HTTP requests. <xref:Sys
+   The JSON for a repository object contains dozens of properties, but only the `Name` property will be deserialized. The serializer automatically ignores JSON properties for which there is no match in the target class. This feature makes it easier to create types that work with only a subset of fields in a large JSON packet.
 
-   The C# convention is to [capitalize the first letter of property names](../../standard/design-guidelines/capitalization-conventions.md), but the `name` property here starts with a lowercase letter because that matches exactly what's in the JSON. Later you'll see how to use C# property names that don't match the JSON property names.
+   The C# convention is to [capitalize the first letter of property names](../../standard/design-guidelines/capitalization-conventions.md).
```
It's worth adding that the `GetFromJsonAsync` uses case-insensitive property names. That's one of the important benefits of using it.

BartoszKlonowski replied:
I added it now in two places, please see the latest commit.

BillWagner reviewed (commented):
This looks great @BartoszKlonowski 

I had a few comments, and then it's ready to merge.

BartoszKlonowski committed:
Revert "Correct the orders of points"

This reverts commit bb708ace22ec7a1c8234e2c481ffb0a08a897a5d.

BartoszKlonowski committed:
Remove leftover sentence

BartoszKlonowski committed:
Mention that GetFromJsonAsync is case-insensitive

This is mentioned in two places - with the C# naming convention, where
it's worth to mention why we keep on having uppercase, and in the Repository
class defining, where we would normally had to make a conversion, but
now we don't because we have case-insensitive method as a tool.

BartoszKlonowski commented on docs/csharp/tutorials/console-webapiclient.md:
```diff
@@ -121,37 +121,33 @@ Use the <xref:System.Net.Http.HttpClient> class to make HTTP requests. <xref:Sys
 
-1. Use the serializer to convert JSON into C# objects. Replace the call to
-<xref:System.Net.Http.HttpClient.GetStringAsync(System.String)> in the `ProcessRepositoriesAsync` method with the following lines:
+2. Use the <xref:System.Net.Http.Json.HttpClientJsonExtensions.GetFromJsonAsync%2A?displayProperty=nameWithType> method to fetch and convert JSON into C# objects. Replace the call to <xref:System.Net.Http.HttpClient.GetStringAsync(System.String)> in the `ProcessRepositoriesAsync` method with the following lines:
```
Didn't know that, thanks! 👍

BillWagner reviewed (approved):
This is great @BartoszKlonowski 

I'll :shipit: as soon as the build finishes.
